File: app/src/main/java/info/firozansari/rxbindingexamples/CountDownActivity.java
package info.firozansari.rxbindingexamples;

import androidx.appcompat.app.AppCompatActivity;
import android.os.Bundle;
import android.widget.Button;

import com.jakewharton.rxbinding2.view.RxView;
import com.jakewharton.rxbinding2.widget.RxTextView;

import java.util.concurrent.TimeUnit;

import butterknife.BindView;
import io.reactivex.Flowable;
import io.reactivex.android.schedulers.AndroidSchedulers;

public class CountDownActivity extends BaseActivity {

    @BindView(R.id.btn_get_code)
    Button btnGetCode;

    @Override
    protected int getLayoutId() {
        return R.layout.activity_count_down;
    }

    @Override
    protected void onViewCreated(Bundle savedInstanceState) {

        addDisposable(RxView.clicks(btnGetCode)
                .throttleFirst(1, TimeUnit.SECONDS)
                .observeOn(AndroidSchedulers.mainThread())
                .doOnNext(o -> RxView.enabled(btnGetCode).accept(false))
                .subscribe(o -> {

                            addDisposable(Flowable.intervalRange(0, 11, 0, 1, TimeUnit.SECONDS)
                                    .observeOn(AndroidSchedulers.mainThread())
                                    .doOnNext(aLong -> {
                                        RxTextView.text(btnGetCode).accept(" " + (10 - aLong) + " ");
                                    })
                                    .doOnComplete(() -> {
                                        RxView.enabled(btnGetCode).accept(true);
                                        RxTextView.text(btnGetCode).accept("Done");
                                    })
                                    .subscribe());
                        }
                )
        );
    }
}

File: app/src/main/java/info/firozansari/rxbindingexamples/CountDownActivity.kt
package info.firozansari.rxbindingexamples

import android.os.Bundle
import android.widget.Button
import android.widget.TextView
import butterknife.BindView
import com.jakewharton.rxbinding2.view.RxView
import io.reactivex.android.schedulers.AndroidSchedulers
import java.util.concurrent.TimeUnit

class CountDownActivity : BaseActivity() {
    @JvmField
    @BindView(R.id.btn_get_code)
    var btnGetCode: Button? = null

    @JvmField
    @BindView(R.id.increment_button)
    var incrementButton: Button? = null

    @JvmField
    @BindView(R.id.my_text_view)
    var myTextView: TextView? = null
    protected override val layoutId: Int
        protected get() = R.layout.activity_count_down

    override fun onViewCreated(savedInstanceState: Bundle?) {
        addDisposable(RxView.clicks(btnGetCode!!)
                .throttleFirst(1, TimeUnit.SECONDS)
                .observeOn(AndroidSchedulers.mainThread())
                .doOnNext { o: Any? -> RxView.enabled(btnGetCode!!).accept(false) }
                .subscribe { o: Any? -> }
        )

        //broadcast clicks into a cumulative increment, and display in TextView
        addDisposable(RxView.clicks(incrementButton!!)
                .map { o: Any? -> 1 }
                .scan(0, { total: Int, next: Int -> total + next })
                .subscribe { i: Int -> myTextView!!.text = i.toString() })
    }
}